### PR TITLE
Friend Profile Integration

### PR DESCRIPTION
**Pull request by Tiewhan**
<h1>Overview</h1>
Part 1 of 2 of the Image Integration into Feisty. This Request deals with integration logic of the profile images for friends.

This is done by adding a new repository that handles calling and decoding of profile images. The View Model and Models have also been updated to handle this chain of events.

<h2>Notes</h2>
<b>Removed certain retain cycles to do with the MVVM architecture.</b>

To fix this each point that had an observer had that observer set to be a weak reference. The observer system was also changed from a dictionary of observers to only a single observer as it is unlikely that a model will have multiple view-models observing it, for example.

### CODE REVIEW

File: Source/CoreClasses/User.swift [hunk 1]
```diff
@@ -20,6 +20,7 @@ public class User {
   let userID: String
```
Comment (Tiewhan) [inline Source/CoreClasses/User.swift:20]: Users now have an image attached. Seemed appropriate to do this as it used to transport the image from service call to being presented on the view

**Review by AkuaQ — APPROVED** [review]

File: Example/Tests/FriendTests/FriendListTests/Mocks/MockFriendListViewModel.swift [hunk 2]
```diff
@@ -13,11 +13,13 @@ class MockFriendListViewModel: FriendListViewModelType {
   
   var isFriendListFound: Bool
   var isFriendListNotFound: Bool
+  var isProfileImageFound: Bool
   
   init() {
     
     isFriendListFound = false
     isFriendListNotFound = false
+    isProfileImageFound = false
```
Comment (Zamo22) [inline Example/Tests/FriendTests/FriendListTests/Mocks/MockFriendListViewModel.swift:22]: Hmm, the init simply sets these 3 items to false. Why not set their initial value as false directly and not need an init.
Edit: I just saw this is a mock, no need to be too strict then, feel free to ignore

File: Source/ServiceCalls/FeistyAPIURLComponents.swift [hunk 3]
```diff
@@ -9,8 +9,10 @@ import Foundation
 
 struct FeistyAPIURLComponents {
   
-  private static let httpProtocol = "https://"
-  private static let domainName = "feisty-server.herokuapp.com"
+  //private static let httpProtocol = "https://"
```
Comment (Zamo22) [inline Source/ServiceCalls/FeistyAPIURLComponents.swift:12]: Remember to delete these if they're not used

**Comment by Zamo22** [comment]
Try to keep the PRs a bit smaller in future, will be a very necessary skill to have at clients

**Review by Zamo22 — APPROVED** [review]

**Pull request merged**